Add overview about web scraping

laughk opened the pull request:
このPRは #9 の出直しのうちの1つです。

「スクレイピングとは何か？」の説明がなくいきなり話が始まるので、
スクレイピングそのものの大枠の説明を追加します。

laughk committed:
スクレイピングの大枠の説明を追加

kashewnuts commented on 2_scraping.md:
```diff
@@ -1,6 +1,16 @@
 # 2.スクレイピング
 
-Pythonでスクレイピングをする時に使えるツールなどのメモです。
+スクレイピングとは Web サイトから自分のほしい情報に絞り込んで収集する技術のことをいいます。  
```
`に`→`を`です。
`スクレイピングとは、 Web サイトから自分のほしい情報を絞り込んで収集する技術のことをいいます。`

kashewnuts commented on 2_scraping.md:
```diff
@@ -1,6 +1,16 @@
 
-Pythonでスクレイピングをする時に使えるツールなどのメモです。
+スクレイピングとは Web サイトから自分のほしい情報に絞り込んで収集する技術のことをいいます。  
+おおまかな処理の流れとしては以下のようなイメージです。
```
`処理の流れとしては以下のイメージです。` と言い切ってください。

kashewnuts commented on 2_scraping.md:
```diff
@@ -1,6 +1,16 @@
+処理 | ライブラリ | ドキュメント
+-|-|-
+Web ページから情報を取得 (図の1,2) | requests | [Requests: HTTP for Humans](http://docs.python-requests.org/en/master/)
+response から抜きたい情報を取り出す (図の3) | BeautifulSoup | [Beautiful Soup Documentation](https://www.crummy.com/software/BeautifulSoup/bs4/doc/)
```
`ほしい情報を取り出す` の方がいいと思います。`抜きたい情報を取り出す` だと意味はわかりますが、一瞬何をいってるかわかりません。scraping-imagineの図も同様に `3. ... ほしい情報を取り出す`, `4. 取り出した情報を利用する` がよいでしょう。

kashewnuts commented:
@laughk 見ました。確認お願いします。

laughk committed:
日本語がおかしかったので修正

ref. https://github.com/pynyumon/pynyumon/pull/12#discussion_r143356364

laughk committed:
ぼかさず言い切る

laughk committed:
「抜きたい情報」ではなく「欲しい情報」という表現にそろえる

scraping-imagine.jpg の変更内容
---------------------------------
図の中の文言を変更

- 「3. response から抜きたい情報を取り出す」->「3. response からほしい情報を取り出す」
- 「4. 情報を抜き出した結果を利用する」->「4. 取り出した情報を利用する」

laughk commented:
@kashewnuts 

丁寧にレビューいただきありがとうごうざいます!!
頂いたコメント内容を反映したので再度確認をお願いします :pray:

kashewnuts commented:
LGTM! 🎉

takanory commented on 2_scraping.md:
```diff
@@ -1,6 +1,16 @@
+
+処理 | ライブラリ | ドキュメント
+-|-|-
+Web ページから情報を取得 (図の1,2) | requests | [Requests: HTTP for Humans](http://docs.python-requests.org/en/master/)
```
ライブラリの名前は Requests かな(先頭大文字)

takanory commented on 2_scraping.md:
```diff
@@ -1,6 +1,16 @@
+処理 | ライブラリ | ドキュメント
+-|-|-
+Web ページから情報を取得 (図の1,2) | requests | [Requests: HTTP for Humans](http://docs.python-requests.org/en/master/)
+response からほしい情報を取り出す (図の3) | BeautifulSoup | [Beautiful Soup Documentation](https://www.crummy.com/software/BeautifulSoup/bs4/doc/)
```
ライブラリの名前は `Beautiful Soup 4` がいいかなー

takanory replied:
厳密に言うと bs4 がやっているのは HTML からのデータの抜き出しで、response 全体は見ていない。まぁ、このままの説明でもいいけど

kashewnuts replied:
日本語版のリンクが復活していたので、 http://kondou.com/BS4/ に直したい

laughk replied:
日本語版のドキュメントは復活しているものの、
以下の点を考慮して直すのではなく参考リンクとして記載しておくのが良さそうだと思いました。

- ドキュメントの内容は 2013-11-19 最終更新と内容が古い
- http://kondou.com/ は個人サイトのようで、再びアクセスできなくなる可能性もある。

kashewnuts replied:
たしかに。古い上にアクセスできなくなる可能性を考えるとこのままでいいかも。

takanory reviewed (commented):
だいたいいいけど、できれば対応していほしいことをコメントに書いた

laughk committed:
ライブラリの名称なので頭文字は大文字にする

laughk committed:
ライブラリの名称の修正

- 公式ドキュメントを見る限る、「Beautiful Soup」が正式
  - see. https://www.crummy.com/software/BeautifulSoup/bs4/doc/
- version 3 と 4 では別のライブラリとして PyPI に登録されていることから「4」まで明記する。

laughk committed:
参考程度に Beautiful Soup 4 の日本語訳版のドキュメントのリンクを紹介

laughk commented:
@takanory @kashewnuts 

コメント頂いた内容を取り込んだので、再度ご確認お願いします :pray:

takanory commented:
![lgtm-f-hands-up](https://user-images.githubusercontent.com/988241/31369479-d3cc64a4-adbf-11e7-89f1-635beeb7feb1.jpg)
